CSAPI/index.js

```diff
@@ -0,0 +1,99 @@
+/*
+Author: Ben Frey
+Original Source: https://www.codingpanel.com/build-a-simple-web-application-using-node-js-mysql/
+All Rights Reserved to the Original Owner
+
+Description: Modified web app that can store events in a simple list by accessing
+on of our MySQL databases (Team Database).
+*/
+
+// Import fs, express, hbs (HTML template format), body-parser
+let fs = require('fs');
+const express = require('express');
+const hbs = require('express-handlebars');
+const bodyParser = require("body-parser");
+
+// Initialize express as app
+const app = express();
+
+// Need to make connection
+const con = require('./models/taskModel');
+
+// User express to populate HTML templates
+app.use(express.static('public'));
+app.engine('hbs', hbs.engine({
+    helpers: {
+        isCompleted: function (status) {
+            if (status == "completed") {
+                return true
+            } else {
+                return false
+            }
+        },
+    },
+    layoutsDir: __dirname + '/views/layouts',
+    defaultLayout: 'main',
+    extname: '.hbs'
+}));
+app.set('view engine', 'hbs');
+app.use(bodyParser.urlencoded({
+    extended: true
+}))
+app.use(bodyParser.json())
+
+app.get('/', (req, res) => {
+    let query = "SELECT * FROM dept";
+    let items = []
+    con.query(query, (err, result) => {
+        if (err) throw err;
+        items = result
+        console.log(items)
+        res.render('index', {
+            items: items
+        })
+    })
+    console.log(items);
+});
+
+app.get('/classes:id', (req, res) => {
+    let query = "SELECT * FROM class";
+    let items = []
+    console.log(req)
+    let condition = req.query.dept_code
+    if (condition){
+        console.log(condition)
+        condArray = condition.split(",");
+        // Add to sql query
+        query = query + " WHERE dept_code = '" + condArray.join("' OR dept_code = '")+ "'";
+      }
+    console.log(query)
+    con.query(query, (err, result) => {
+        if (err) throw err;
+        items = result
+        console.log(items)
+        res.render('index', {
+            items: items
+        })
+    })
+    console.log(items);
+});
+
+app.post('/classes', (req, res) => {
+    console.log(req.body)
+    let query = "INSERT INTO class (dept_code, class_num, class_name) VALUES ?";
```

Comment:
does this put `data` (which goes into the `?`) into parentheses? I think you need the parentheses for the query to work correctly

Reply:
I haven't had issues with it in practice. I believe the con.query operation handles the combining of the query with the [data], but can look further into it.